src/core/server/http/http_server.ts

```diff
@@ -104,9 +105,16 @@ export class HttpServer {
     return this.server !== undefined && this.server.listener.listening;
   }
 
-  private registerRouter(router: IRouter) {
+  private registerRouter(router: IRouter, allowAfterListening = false) {
```

Comment:
optional: could you create a dedicated method to implement this functionality? The route registration lifecycle is absolutely different.

Reply:
Resolved in [`fcc1853` (#79012)](https://github.com/elastic/kibana/pull/79012/commits/fcc18533523cb94926a41af422cbbfa7c7642c55)